examples/showcase/tracked-controls/js/components/collider.js

```diff
@@ -0,0 +1,109 @@
+/* global AFRAME, THREE */
+
+/**
+ * It implements AABB collision detection for entities with a mesh.
+ * (https://en.wikipedia.org/wiki/Minimum_bounding_box#Axis-aligned_minimum_bounding_box)
+ * It sets the specified state on the intersected entities.
+ *
+ * @property {string} Selector of the entities to test for collision.
+ * @property {string} State to set on collided entities.
+ *
+ */
+AFRAME.registerComponent('collider', {
+  schema: {
+    objects: {default: ''},
+    state: {default: 'collided'}
+  },
+
+  init: function () {
+    this.els = [];
+    this.collisions = [];
+    this.elMax = new THREE.Vector3();
+    this.elMin = new THREE.Vector3();
+  },
+
+  update: function () {
+    this.updateEls();
+  },
+
+  /**
+   * Update list of entities to test for collision.
+   */
+  updateEls: function () {
+    var data = this.data;
+    var objectEls;
+
+    // Push entities into list of els to intersect.
+    if (data.objects) {
+      objectEls = this.el.sceneEl.querySelectorAll(data.objects);
+    } else {
+      // If objects not defined, intersect with everything.
+      objectEls = this.el.sceneEl.children;
+    }
+    // Convert from NodeList to Array
+    this.els = Array.prototype.slice.call(objectEls);
+  },
+
+  tick: (function () {
+    var boundingBox = new THREE.Box3();
+    return function () {
+      var collisions = [];
+      var el = this.el;
+      var mesh = el.getObject3D('mesh');
+      var self = this;
+      // No mesh no collisions
+      if (!mesh) { return; }
+      // Update the bounding box to account for rotations and
+      // position changes.
+      updateBoundingBox();
+      // Update collisions.
+      this.els.forEach(intersect);
+      // Emit events.
+      collisions.forEach(hit);
+      // no collisions.
+      if (collisions.length === 0) { self.el.emit('hit', {el: null}); }
```

Comment:
I'm still not sure about prepending the component name to the event name. It feels heavy and error prone. It would maybe be better to have an `emit` API on the component that adds the component name to the event details.